Makefile

```diff
@@ -90,6 +90,10 @@ test: export CXXFLAGS := --this_flag_does_not_exist # Env removed in text_gcovr.
 test: export GCOV := $(GCOV)
 
 test:
+ifneq ($(wildcard $(dir $(word 1,$(abspath $(MAKEFILE_LIST))))gcovr/gcovr/tests/linked/subdir/loop),)
+	# Remove directory because test is not running if it exists
+	rm -Rf $(dir $(word 1,$(abspath $(MAKEFILE_LIST))))gcovr/gcovr/tests/linked/subdir/loop
+endif
```

Comment:
Why does this have to be done in the top-level makefile? Can't this be moved to the Makefile for that specific test?

Reply:
Because pytest collection phase has an infinite loop if this directory exists from a previous run. I'll try to configure pytest to ignore the directory.

Reply:
Oh no! That is a *very good* reason to keep this code here. I didn't understand the context from the “test is not running” comment, so expanding the comment to mention pytest collection + infinite loops would be perfect.

I'm also wondering if the combined Make/shell could be simplified a bit. I'd guess a pure-shell command might be clearer:

```
@ cursed_directory=...; if test -d "$$cursed_directory'; then rm -Rf "$$cursed_directory"; fi
```

<sub>`@` to avoid this lengthy command from being printed out by Make, `$$` for escaping, not sure about the correct test flag (`-d`, `-L`, `-e`?)</sub>

Could you also explain why an absolute path is needed here? Makefiles are effectively guaranteed to be only called from the correct directory – whoever uses `make -f` has issues. <sub>(Disclaimer: our `gcovr/tests/Makefile` uses this flag, but it seems unnecessary. I didn't even know we had that file!</sub>

Reply:
I think this was because the wildcard function needs an absolute path.
In the test directory there is also a makefile which calls the clean target for all tests. Maybe calling this before pytest is better. Also adding the mentioned comment why it is called.

Reply:
I've added pytest options to setup.cfg and know the directory is ignored.